src/users_router.rs

```diff
@@ -7,16 +7,68 @@ extern crate iron;
 extern crate router;
 
 use self::iron::{AfterMiddleware, headers, status};
+use self::iron::method::Method;
 use self::iron::method::Method::*;
 use self::iron::prelude::*;
 use self::router::Router;
 use self::unicase::UniCase;
 
+type Endpoint = (Method, &'static[&'static str]);
+
 struct CORS;
 
+impl CORS {
+    // Only endpoints listed here will allow CORS.
+    // Endpoints containing a variable path part can use '*' like in:
+    // &["users", "*"]
+    pub const ENDPOINTS: &'static[Endpoint] = &[
+        (Method::Post,      &["invitations"]),
+        (Method::Get,       &["invitations"]),
+        (Method::Delete,    &["invitations"]),
+        (Method::Post,      &["users"]),
+        (Method::Get,       &["users"]),
+        (Method::Put,       &["users", "*"]),
+        (Method::Post,      &["users", "*"]),
+        (Method::Post,      &["recoveries", "*"]),
+        (Method::Get,       &["recoveries", "*", "*"]),
+        (Method::Get,       &["permissions"]),
+        (Method::Get,       &["permissions", "*"]),
+        (Method::Get,       &["permissions", "*", "*"]),
+        (Method::Get,       &["permissions", "_", "*"]),
+        (Method::Put,       &["permissions", "*", "*"]),
+    ];
+}
+
 impl AfterMiddleware for CORS {
-    fn after(&self, _: &mut Request, mut res: Response)
+    fn after(&self, req: &mut Request, mut res: Response)
         -> IronResult<Response> {
+
+        let mut is_cors_endpoint = false;
+        for endpoint in CORS::ENDPOINTS {
+            let (ref method, path) = *endpoint;
+            if req.method != *method {
+                continue;
+            }
+            if path.len() != req.url.path.len() {
+                continue;
+            }
+            for (i, path) in path.iter().enumerate() {
+                is_cors_endpoint = false;
+                if req.url.path[i] != path.to_string() &&
+                   "*".to_string() != path.to_string() {
+                    break;
+                }
+                is_cors_endpoint = true;
+            }
+            if is_cors_endpoint {
+                break;
+            }
+        }
+
+        if !is_cors_endpoint {
```

Comment:
Nit: you can do `if is_cors_endpoint {` and then put lines 72 - 77 inside the if-statement.